transports/websocket-websys/src/web_context.rs

```diff
@@ -0,0 +1,62 @@
+use wasm_bindgen::{prelude::*, JsCast};
+
+/// Web context that abstract the window vs web worker API
+#[derive(Debug)]
+pub(crate) enum WebContext {
+    Window(web_sys::Window),
+    Worker(web_sys::WorkerGlobalScope),
+}
+
+impl WebContext {
+    pub(crate) fn new() -> Self {
+        #[wasm_bindgen]
+        extern "C" {
+            type Global;
+
+            #[wasm_bindgen(method, getter, js_name = Window)]
+            fn window(this: &Global) -> JsValue;
+
+            #[wasm_bindgen(method, getter, js_name = WorkerGlobalScope)]
+            fn worker(this: &Global) -> JsValue;
+        }
+
+        let global: Global = js_sys::global().unchecked_into();
+
+        if !global.window().is_undefined() {
+            Self::Window(global.unchecked_into())
+        } else if !global.worker().is_undefined() {
+            Self::Worker(global.unchecked_into())
+        } else {
+            panic!("Only supported in a browser or web worker");
```

Comment:
I think I'd prefer `new` to return an `Option` and leave the panicking to upper layers.